packages/myst-cli/src/build/cff.ts

```diff
@@ -0,0 +1,323 @@
+    copyright: frontmatter.copyright,
+    identifiers: frontmatter.doi
+      ? [
+          {
+            type: 'doi',
+            value: frontmatter.doi,
+          },
+        ]
+      : undefined,
+    editors: frontmatter.editors
+      ?.map((id) => {
+        const editor = frontmatter.contributors?.find((contrib) => contrib.id === id);
+        if (!editor) return undefined;
+        return authorToCFF(editor, frontmatter.affiliations);
+      })
+      .filter((editor): editor is PersonCFF | EntityCFF => !!editor),
+    end: typeof last_page === 'number' ? last_page : undefined,
+    issue: typeof issue === 'number' ? issue : undefined,
+    'issue-title': typeof issue === 'string' ? issue : undefined,
+    journal: frontmatter.venue?.title,
+    keywords: frontmatter.keywords,
+    license: license?.id,
+    'license-url': license?.url,
+    pages:
+      typeof last_page === 'number' && typeof first_page === 'number'
+        ? last_page - first_page + 1
+        : undefined,
+    start: typeof first_page === 'number' ? first_page : undefined,
+    url: frontmatter.source,
+    volume: typeof volume === 'number' ? volume : undefined,
+    'volume-title': typeof volume === 'string' ? volume : undefined,
+    repository: frontmatter?.github,
+  };
+}
+
+export async function runCffExport(
+  session: ISession,
+  sourceFile: string,
+  exportOptions: ExportWithOutput,
+  opts?: ExportFnOptions,
+) {
+  const toc = tic();
+  const { output, articles } = exportOptions;
+  const { clean, projectPath, extraLinkTransformers } = opts ?? {};
+  const article = articles[0];
+  const state = session.store.getState();
+  let frontmatter: PageFrontmatter | undefined;
+  let abstract: string | undefined;
+  if (projectPath && selectors.selectLocalConfigFile(state, projectPath) === sourceFile) {
+    frontmatter = selectors.selectLocalProjectConfig(state, projectPath);
+    const rawAbstract = frontmatter?.parts?.abstract?.join('\n\n');
+    if (rawAbstract) {
+      const abstractAst = parseMyst(session, rawAbstract, sourceFile);
+      abstract = toText(abstractAst);
+    }
+  } else if (article.file) {
+    const [content] = await getFileContent(session, [article.file], {
+      projectPath,
+      imageExtensions: KNOWN_IMAGE_EXTENSIONS,
+      extraLinkTransformers,
+    });
+    frontmatter = content.frontmatter;
+    const abstractMdast = extractPart(content.mdast, 'abstract');
+    if (abstractMdast) abstract = toText(abstractMdast);
+  }
+  if (!frontmatter) return { tempFolders: [] };
+  if (clean) cleanOutput(session, output);
+  const vfile = new VFile();
+  vfile.path = output;
+  if (path.basename(output) !== 'CITATION.cff') {
+    fileWarn(
+      vfile,
+      `Invalid Citation File Format filename ${path.basename(output)} - CFF requires filename 'CITATION.cff'`,
+    );
+  }
+  const cff = {
+    ...frontmatterToCFF(frontmatter, abstract),
+    ...exportOptionsToCFF(exportOptions),
```

Comment:
This line allows authors to add (or override) additional CFF info to the export definition, e.g.
```yaml
exports:
  - format: cff
    message: If you cite this article, thank you!!
```